### PR TITLE
Fix bad links - System namespace

### PR DESCRIPTION
**Pull request by gewarren**
Contributes to #10311.

### CODE REVIEW

File: xml/System/Object.xml [hunk 1]
```diff
@@ -422,7 +422,7 @@ For an additional example that overrides the <xref:System.Object.Finalize%2A> me
 
 -   If two objects compare as equal, the <see cref="M:System.Object.GetHashCode" /> method for each object must return the same value. However, if two objects do not compare as equal, the <see cref="M:System.Object.GetHashCode" /> methods for the two objects do not have to return different values.
 
--   The <see cref="M:System.Object.GetHashCode" /> method for an object must consistently return the same hash code as long as there is no modification to the object state that determines the return value of the object's [System.Object.Equals](xref:System.Object.Equals*) method. Note that this is true only for the current execution of an application, and that a different hash code can be returned if the application is run again.
+-   The <see cref="M:System.Object.GetHashCode" /> method for an object must consistently return the same hash code as long as there is no modification to the object state that determines the return value of the object's <see href="xref:System.Object.Equals*">System.Object.Equals</see> method. Note that this is true only for the current execution of an application, and that a different hash code can be returned if the application is run again.
```
Comment (gewarren) [inline xml/System/Object.xml:425]: Need to check if this works with xref links or if we need to convert to cref.

File: xml/System/DateTime.xml [hunk 2]
```diff
@@ -6036,8 +6036,8 @@ The following example parses an array of date strings by using the conventions o
         <Parameter Name="styles" Type="System.Globalization.DateTimeStyles" />
       </Parameters>
       <Docs>
-        <param name="s">A string that contains a date and time to convert. See [The string to parse](#StringToParse) for more information.</param>
-        <param name="provider">An object that supplies culture-specific formatting information about <paramref name="s" />.  See [Parsing and cultural conventions](#Culture)</param>
+        <param name="s">A string that contains a date and time to convert. See <see href="#StringToParse">The string to parse</see> for more information.</param>
```
Comment (gewarren) [inline xml/System/DateTime.xml:6039]: Does it work for bookmarks?

File: xml/System/Delegate.xml [hunk 3]
```diff
@@ -292,7 +292,7 @@ For examples, see [Supplemental API remarks for System.Delegate.CreateDelegate](
 
  <paramref name="method" /> is <see langword="null" />.</exception>
         <exception cref="T:System.ArgumentException">
-          <paramref name="target" /> is not a <see langword="RuntimeType" />. See [Runtime Types in Reflection](/previous-versions/dotnet/netframework-4.0/ms172329(v=vs.100)).
+          <paramref name="target" /> is not a <see langword="RuntimeType" />. See <see href="/previous-versions/dotnet/netframework-4.0/ms172329(v=vs.100">Runtime Types in Reflection</see>).
```
Comment (gewarren) [inline xml/System/Delegate.xml:295]: Okay need to be smarter about parentheses within link URL.

File: xml/System/DateTime.xml (hunk 2 above)
Comment (gewarren) [reply xml/System/DateTime.xml:6039]: Yes it does!

File: xml/System/Object.xml (hunk 1 above)
Comment (gewarren) [reply xml/System/Object.xml:425]: Yessiree.

**Review by BillWagner — APPROVED** [review]
This LGTM @gewarren 

Let's :shipit:

**Pull request merged**